src/Lib/CrashDetector.php

```diff
@@ -0,0 +1,108 @@
+
+/*
+ Copyright (c) 2023, Manticore Software LTD (https://manticoresearch.com)
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License version 2 or any later
+ version. You should have received a copy of the GPL license along with this
+ program; if you did not, you can find it at http://www.gnu.org/
+ */
+
+namespace Manticoresearch\Buddy\Lib;
+
+use Manticoresearch\Buddy\Network\ManticoreClient\Settings;
+
+class CrashDetector {
+	/**
+	 * Initialize crash detector by using manticore settings
+	 * @param Settings $settings
+	 * @return void
+	 */
+	public function __construct(protected Settings $settings) {
+	}
+
+	/**
+	 * The method check searchd.log and try to find if we crashed last time
+	 * and send metric that increments crash counter
+	 * @return bool
+	 * 	we return false even when we cannot detect that we had crash
+	 */
+	public function hadCrash(): bool {
+		// If we have no log set, skip it
+		if (!$this->settings->searchdLog) {
+			return false;
+		}
+
+		// Let's parse searchd.log now and find out last crash
+		$fp = fopen($this->settings->searchdLog, 'rb');
+		if (!$fp) {
+			return false;
+		}
+		$hadCrash = static::hadCrashInLog($fp);
+		fclose($fp);
+
+		return $hadCrash;
+	}
+
+	/**
+	 * Get resource to the file and read it from the end to try to find crash log
+	 * @param mixed $fp
+	 * @return bool
+	 */
+	protected static function hadCrashInLog(mixed $fp): bool {
+		/** @var resource $fp */
+		$hadCrash = false;
+		fseek($fp, 0, SEEK_END);
+		$pos = ftell($fp) - 1;
+		$starts = 0;
+		$line = '';
+		while ($pos > 0) {
+			$chr = fgetc($fp);
+			fseek($fp, --$pos);
+			if ($chr === PHP_EOL) {
+				[$starts, $hadCrash] = static::validateLogLine($line, $starts);
+				if ($hadCrash || $starts >= 2) {
+					break;
+				}
+				$line = '';
+				continue;
+			}
+			$line = $chr . $line;
+		}
+
+		return $hadCrash;
+	}
+
+	/**
+	 * We just split in small functions just because of cognitive complexity is high
+	 * This function just look into line and detect if we had crash and return info
+	 * @param string $line
+	 * @param int $starts
+	 * @return array{0:int,1:bool}
+	 *  array with values of starts and hadCrash
+	 */
+	protected static function validateLogLine(string $line, int $starts): array {
+		// We keep it to look for 2nd one, because first one
+		// from the end of the file – is current daemon launch process
+		if (str_contains($line, 'starting daemon version')) {
+			++$starts;
+		}
+
+		// Try find crash in lines last launch belongs to
+		if ($starts === 1) {
+			// We also can stop when we found crash
+			// @see: https://github.com/manticoresoftware/manticoresearch/blob/master/src/searchd.cpp#L960
+			// We are checking from the end, so check ending of crash that looks like:
+			// --- crashed XXX request dump ---
+			// …
+			// backtrace
+			// --- request dump end ---
+			if (str_contains($line, '--- request dump end ---')) {
```

Comment:
I'm afraid some crashes may not have this line. It may be better to use "- CRASH" instead.